addon/ng2/commands/test.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+var chalk        = require('chalk');
+var Command      = require('ember-cli/lib/models/command');
+var Promise      = require('ember-cli/lib/ext/promise');
+var Project      = require('ember-cli/lib/models/project');
+var SilentError        = require('silent-error');
+var validProjectName   = require('ember-cli/lib/utilities/valid-project-name');
+var normalizeBlueprint = require('ember-cli/lib/utilities/normalize-blueprint-option');
+
+var TestCommand = require('ember-cli/lib/commands/test');
+var win = require('ember-cli/lib/utilities/windows-admin');
+var path = require('path');
+
+// require dependencies within the target project
+function requireDependency (root, moduleName) {
+  var packageJson = require(path.join(root, 'node_modules', moduleName, 'package.json'));
+  var main = path.normalize(packageJson.main);
+  return require(path.join(root, 'node_modules', moduleName, main));
+}
+
+module.exports = TestCommand.extend({
+
+    init: function(){
+      if (!this.karmaServer) {
+        var karma = requireDependency(this.project.root, 'karma');
+        var karmaConfig = path.join(this.project.root, 'karma.conf');
+        this.karmaServer = new karma.Server({configFile: karmaConfig});
+      }
+    },
+
+    run: function(commandOptions, rawArgs) {
+      var BuildTask = this.tasks.Build;
+      var buildTask = new BuildTask({
+        ui: this.ui,
+        analytics: this.analytics,
+        project: this.project
+      });
+
+      var buildCommandOptions = {
+        environment: 'development',
+        outputPath: 'dist/',
+        watch: false
+      };
+
+      var self = this;
+      return win.checkWindowsElevation(this.ui)
+        .then(function() {
+          return buildTask.run(buildCommandOptions);
```

Comment:
we should support passing through karma command line flags like `--single-run` into the server, otherwise I'm not sure how we'll use `ng test` on CI